### ISSUE TITLE
Leaf config doesn't override higher up config?

### ISSUE THREAD

**Issue by cymen** [rt2zz/redux-persist#824]
I had a state tree like this:

state:
```
{
  auth: {
    token: 'some_token',
    error: null,
  },
 ...
}
```

I started out with this config:

```
const persistConfig = {
  key: 'root',
 whitelist: ['auth'],
 storage,
};
```

But then I realized I didn't want to persist `auth.error` so I made a more specific config for it:

```
const authPersistConfig = {
  key: 'auth',
  blacklist: ['error'],
  storage,
};
```

Interesting thing was the higher up whitelist overrode the more specific leaf config so `auth.error` continued to be persisted. Eventually, I guessed this was the case and got rid of my `persistConfig` and just used the config for the `auth` leaf.

Shouldn't a local leaf config `blacklist` override a higher up `whitelist`?

**Comment by peterlazar1993** [comment]
The last block of code looks same as your initial setup, that is parent whitelists 'auth', child blacklists 'error' in effect blacklisting 'auth.error'. But you said this is not working for you. 
So which is it, does the last code block or not?

I am also facing a similar problem, I'm trying to blacklist a key of one of my reducers, the parent persistConfig blacklists 2 other reducers, I want my child persistConfig to blacklist a key which does not work. Which leads me to question if the nested persisters work? Is there an example I can run to test this?

**Comment by atlanteh** [comment]
I can confirm this doesn't work for me either

**Comment by bruno-edo** [comment]
Okay, think I figured it out. So, in @cymen code his configs look like this:

```javascript
const rootPersistConfig = {
  key: 'root',
  storage: storage,
  whitelist: ['auth']
}

const authPersistConfig = {
  key: 'auth',
  storage: storage,
  blacklist: ['error']
}
```

Which don't work and are the same thing I'm using in my project. 

If we add a `blacklist` key to  `rootPersistConfig` things should work as intended (I tested it in my app and was able to stop persisting one specific key from my reducer).

This should work:

```javascript
const rootPersistConfig = {
  key: 'root',
  storage: storage,
  whitelist: ['auth'],
  blacklist: ['auth'],
}

const authPersistConfig = {
  key: 'auth',
  storage: storage,
  blacklist: ['error']
}
```
This behaviour is really weird and I believe a better explanation on how nested persist work (and how to implement them) should be added to the docs, the example in the README is really vague and leads to confusion when using the feature.

**Comment by uuau99999** [comment]
Same issue here. A more detailed explanation about the nested persist config is expected in doc.

**Comment by microcipcip** [comment]
@uuau99999 I had so many problems with this that I ditched this lib and wrote my own.

**Comment by uuau99999** [comment]
@microcipcip That's awesome. I now use the work around offered by @bruno-edo now and it dose work. But it's a little bit weird.

**Comment by microcipcip** [comment]
@uuau99999 Yeah, it is strange that a 6K stars project doesn't handle basic blacklist properly.
